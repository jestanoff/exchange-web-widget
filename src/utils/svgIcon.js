import React from 'react';
import pure from 'recompose/pure';

export default (svg, displayName) => {
  const Icon = pure(props => React.cloneElement(svg, props));
  Icon.displayName = displayName;

  return Icon;
};
