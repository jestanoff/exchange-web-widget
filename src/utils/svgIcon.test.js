import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import svgIcon from './svgIcon';

describe('svgIcon', () => {
  test('should return Icon react component', () => {
    const svg = (
      <svg>
        <patch d="M1" />
      </svg>
    );
    const Icon = svgIcon(svg, 'icon');
    const { container } = render(<Icon foo="bar" />);

    expect(container).toMatchSnapshot();
  });
});
