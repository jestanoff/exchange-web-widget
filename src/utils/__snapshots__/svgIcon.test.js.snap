// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgIcon should return Icon react component 1`] = `
<div>
  <svg
    foo="bar"
  >
    <patch
      d="M1"
    />
  </svg>
</div>
`;
